### PR TITLE
Fix/12957 risk level change

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Fixes the calculation of risk level change.
Before that fix it was possible to have an increased risk change without having an actual increased risk.

For example, without the fix, this set of risks returned risk change: .increased (the correct change is .unchanged(.high))
currentENFRisk: .high, previousENFRisk: .high, currentCheckinRisk: .low, previousCheckinRisk: .high,

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12627

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Services/Risk/Risk.swift [hunk 1]
```diff
@@ -100,25 +157,24 @@ extension Risk {
...
-		self.init(
-			level: totalRiskLevel,
-			details: details,
-			riskLevelChange: riskLevelChange
-		)
+		
+		return details
+	}
+	
+	// Merge the results from both risk calculation. For each date, the higher risk level is used.
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/Risk/Risk.swift:164]:
```suggestion
	// Merge the results from both risk calculation. For each date, the highest risk level is used.
```

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/Risk/__tests__/RiskTests.swift [hunk 2]
```diff
@@ -114,6 +114,73 @@ class RiskTests: CWATestCase {
 		XCTAssertEqual(risk.details.mostRecentDateWithRiskLevel, today)
 		XCTAssertEqual(risk.details.numberOfDaysWithRiskLevel, 1)
 	}
+	
+	func test_Given_PreviousENFRiskLow_And_PreviousCheckinRiskLow_When_ENFRiskHigh_And_CheckinRiskLow_Then_ChangeIsIncreased() {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/Risk/__tests__/RiskTests.swift:118]: I think the name is outdated

File: src/xcode/ENA/ENA/Source/Services/Risk/Risk.swift [hunk 3]
```diff
@@ -74,21 +119,33 @@ extension Risk {
...
-		Log.debug("[Risk] totalRiskLevel: \(totalRiskLevel)", log: .riskDetection)
-
+		
+		return totalRiskLevel
+	}
+	
+	private static func riskDetails(
+		enfRiskCalculationResult: ENFRiskCalculationResult,
+		checkinCalculationResult: CheckinRiskCalculationResult,
+		risk: RiskLevel
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/Risk/Risk.swift:129]:
```suggestion
		riskLevel: RiskLevel
```

File: src/xcode/ENA/ENA/Source/Services/Risk/Risk.swift [hunk 4]
```diff
@@ -54,16 +54,61 @@ extension Risk {
...
+		if previousTotalRiskLevel == totalRiskLevel {
+			riskLevelChange = .unchanged(totalRiskLevel)
+		} else {
+			riskLevelChange = previousTotalRiskLevel > totalRiskLevel ? .decreased : .increased
+		}
+		
+		Log.debug("[Risk] riskLevelChange: \(riskLevelChange)", log: .riskDetection)
 
-		Log.debug("[Risk] tracingRiskLevelPerDate: \(tracingRiskLevelPerDate)", log: .riskDetection)
-		Log.debug("[Risk] checkinRiskLevelPerDate: \(checkinRiskLevelPerDate)", log: .riskDetection)
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/Risk/Risk.swift:61]: Would it make sense to still log these somewhere, in case we need to debug later?

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Services/Risk/__tests__/RiskTests.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Risk/__tests__/RiskTests.swift:118]: Right. Renamed.

File: src/xcode/ENA/ENA/Source/Services/Risk/Risk.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Risk/Risk.swift:129]: Renamed

File: src/xcode/ENA/ENA/Source/Services/Risk/Risk.swift (hunk 4 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Risk/Risk.swift:61]: Good point. Added logs.

**Review by nickguendling — APPROVED** [review]

**Pull request merged**